cni/pkg/util/podutil.go

```diff
@@ -39,36 +41,46 @@ var annotationRemovePatch = []byte(fmt.Sprintf(
 	constants.AmbientRedirection,
 ))
 
+// TODO: we should use the upstream istio version of this function.
 // PodRedirectionEnabled determines if a pod should or should not be configured
 // to have traffic redirected thru the node proxy.
 func PodRedirectionEnabled(namespace *corev1.Namespace, pod *corev1.Pod) bool {
-	if namespace.GetLabels()[constants.DataplaneMode] != constants.DataplaneModeAmbient {
+	nsLabels := namespace.GetLabels()
+	if nsLabels == nil || nsLabels[constants.DataplaneMode] != constants.DataplaneModeAmbient {
 		// Namespace does not have ambient mode enabled
 		return false
 	}
 	if podHasSidecar(pod) {
 		// Ztunnel and sidecar for a single pod is currently not supported; opt out.
 		return false
 	}
-	if pod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionDisabled {
+	if pod.Annotations != nil && pod.Annotations[constants.AmbientRedirection] == constants.AmbientRedirectionDisabled {
 		// Pod explicitly asked to not have redirection enabled
 		return false
 	}
 	return true
 }
 
 func podHasSidecar(pod *corev1.Pod) bool {
-	if _, f := pod.Annotations[annotation.SidecarStatus.Name]; f {
+	annotations := pod.GetAnnotations()
+	if annotations == nil {
+		return false
+	}
+	if _, f := annotations[annotation.SidecarStatus.Name]; f {
 		return true
 	}
 	return false
 }
 
-func ztunnelPod(pod *corev1.Pod) bool {
-	return pod.GetLabels()["app"] == "ztunnel"
+func IsZtunnelPod(systemNs string, pod *corev1.Pod) bool {
+	labels := pod.GetLabels()
```

Comment:
nit: this check seems not needed, you can read from nil without issue

Reply:
yeah and we do this in a few other spots too, ty.